modules/frontend/search_sharder.go

```diff
@@ -136,7 +136,9 @@ func (s *asyncSearchSharder) backendRequests(ctx context.Context, tenantID strin
 		return
 	}
 
-	blocks := blockMetasForSearch(s.reader.BlockMetas(tenantID), start, end, backend.DefaultReplicationFactor)
+	startT := time.Unix(int64(start), 0)
```

Comment:
Maybe we could move this to `backendRange`
```go
func backendRange(start, end uint32, queryBackendAfter time.Duration) (time.Time, time.Time) 
```
It could also simplify the logic inside that method:

```go
if end.after(backendAfter)
```

Reply:
i'd rather leave `backendRange` alone. it's internally consistent b/c it only ever deals with epoch seconds as its owned by the search path. we could chase this time.Time change quite a ways, but i'd like to keep the changes small to address the issue.